File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    correct = 0
    cnt_0 = lottos.count(0) 
    
    rank = [6, 6, 5, 4, 3, 2, 1]
    
    for lotto_num in lottos:
        if lotto_num in win_nums:
            correct += 1

    return rank[correct + cnt_0], rank[correct]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    
    public int[] solution(int[] lottos, int[] win_nums) {
        int correct = 0;
        int cnt0 = 0;
        
        Set<Integer> winSet = new HashSet<>();
        for (int num : win_nums) {
            winSet.add(num);
        }
        
        for (int lotto : lottos) {
            if (lotto == 0) {
                cnt0++;
            } else if (winSet.contains(lotto)) {
                correct++;
            }
        }
        
        int[] rank = {6, 6, 5, 4, 3, 2, 1};
        
        return new int[] {rank[correct + cnt0], rank[correct]};
    }
}